lib/credentials/shared_ini_file_credentials.js

```diff
@@ -48,11 +51,11 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
 
     this.filename = options.filename;
     this.profile = options.profile || process.env.AWS_PROFILE || 'default';
-    this.get(function() {});
```

Comment:
Short of creating a separate credential provider, I don't think there's a way we can safely change this. That would however mean reading the same file twice, but also eliminates the constructor parameter to disable assume role.